File: easy/maximum-subarray.js
// https://leetcode.com/problems/maximum-subarray/submissions/

var maxSubArray = function(nums) {
    let globalMax = nums[0];
    let prev = 0;
    for (const num of nums) {
        prev = Math.max(prev + num, num);
        globalMax = Math.max(prev, globalMax);
    }
    return globalMax;
};

File: easy/maximum-subarray.py
# since there are negative numbers
# currMax might be smaller than max_sum
# since negative isnt added to max_sum
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
      max_sum = nums[0]
      currMax = nums[0]
      for num in nums[1:]:a
        currMax = max(currMax + num, num)
        if currMax > max_sum:
          max_sum = currMax
      return max_sum